### PR TITLE
add `silicon`

### PR DESCRIPTION
**Pull request by moritzwilksch**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by moritzwilksch** [comment]
@conda-forge/help-rust , this is ready for review

File: recipes/silicon/conda_build_config.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+MACOSX_DEPLOYMENT_TARGET:  # [osx and x86]
```
Comment (xhochy) [inline recipes/silicon/conda_build_config.yaml:1]: Please remove this. 10.13 is the new global pin

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/silicon/conda_build_config.yaml (hunk 1 above)
Comment (moritzwilksch) [reply recipes/silicon/conda_build_config.yaml:1]: ahh right, do i have to re-render instead? Didn't build without this, maybe because the PR is quite old

Comment (xhochy) [reply recipes/silicon/conda_build_config.yaml:1]: There is no rerender on staged-recipes. You probably need to merge/rebase on main.

**Comment by xhochy** [comment]
This will need https://github.com/conda-forge/staged-recipes/pull/26449

File: recipes/silicon/build.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+set -ex
+
+# Bundle all library licenses
+cargo-bundle-licenses \
+  --format yaml \
+  --output ${SRC_DIR}/THIRDPARTY.yml
+
+cargo install --locked --root "$PREFIX" --path .
```
Comment (xhochy) [inline recipes/silicon/build.sh:10]:
```suggestion
cargo install --no-track --locked --root "$PREFIX" --path .
```

**Review by xhochy — APPROVED** [review]

**Pull request merged**